content/en/docs/tasks/debug-application-cluster/debug-running-pod.md

```diff
@@ -172,20 +170,167 @@ Use `kubectl delete` to remove the Pod when you're finished:
 kubectl delete pod ephemeral-demo
 ```
 
-<!--
-Planned future sections include:
+## Debugging using a copy of the Pod
+
+Sometimes Pod configuration options make it difficult to troubleshoot in certain
+situations. For example, `kubectl exec` won't work with a container image that
+doesn't include a shell or if an application crashes on startup. In these
+situations you can use `kubectl debug` to create a copy of the Pod with
+configuration values changed to aid debugging.
+
+### Copying a Pod while adding a new container
+
+Adding a new container can be useful when your application is running but not
+behaving as you expect and you'd like to add additional troubleshooting
+utilities to the Pod.
+
+For example, maybe your application's container images is built on busybox
+but you need debugging utilities not included in busybox. You can simulate
+this scenario using `kubectl run`:
+
+```shell
+kubectl run myapp --image=busybox --restart=Never -- sleep 1d
+```
+
+Run this command to create a copy of `myapp` named `myapp-copy` that adds a
+new ubuntu container for debugging.
+
+```shell
+kubectl debug myapp -it --image=ubuntu --share-processes --copy-to=myapp-debug
+```
+
+```
+Defaulting debug container name to debugger-w7xmf.
+If you don't see a command prompt, try pressing enter.
+root@myapp-debug:/#
+```
+
+There are a few things worth noting in this invocation of `kubectl debug`:
+
+* `kubectl debug` automatically generates a container name  if you don't choose
+  one using the `--container` flag.
+* The `-i` flag causes `kubectl debug` to attach to the new container by
+  default.  You can prevent this by specifying `--attach=false`. If your session
+  becomes disconnected you can reattach using `kubectl attach`.
+* The `--share-processes` causes the Pod to use one process namespace so that
```

Comment:
nit: Line 210, extra space before `if`
nit: The `--share-processes` flag
nit: ... causes the Pod to use one process per namespace. You can identify other processes from other
containers using `ps`.

Reply:
Fixed the first two, but the third suggestion is incorrect. I've updated the text to:

> The `--share-processes` flag causes the Pod to use a single process namespace. You can identify other processes from other containers using `ps`.

I'm not sure if this addresses your initial concern, though.

Reply:
OK. I am not sure what exactly is a "single process namespace"?
Single process + one namespace or ?

Reply:
Under the hood it sets the `shareProcessNamespace` option in [v1.PodSpec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#podspec-v1-core). A "process namespace" is how the kernel keeps application processes in containers separate from processes in other containers and the larger system.

By default every container gets a separate "process namespace" so that there's no visibility to processes that are running in different containers. That makes debugging across container boundaries more difficult.

When one uses `--share-processes`, the kubelet will use one "process namespace" for all of the containers in the Pod so that processes from all of the containers in that Pod are visible to each other.

Reply:
Great. Thanks for the explanation!

Reply:
@kbhawkey Do you think I should rephrase/expand on this concept in the text or is it good as is?

Reply:
I'll read through the page ...